osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/securitygroup/SecurityGroupMemberDeleteTask.java

```diff
@@ -160,4 +141,64 @@ public String getName() {
         return LockObjectReference.getObjectReferences(this.sgm.getSecurityGroup());
     }
 
+    private void deleteFromDB(EntityManager em , VMPort port) {
+        this.log.info("Removing port " + port);
+        VM vm = port.getVm();
+        if (vm != null) {
+            vm.removePort(port);
+            port.setVm(null);
+        }
+
+        Network network = port.getNetwork();
+        if (network != null) {
+            network.removePort(port);
+        }
+
+        for (DistributedApplianceInstance dai : port.getDais()) {
+            dai.removeProtectedPort(port);
+            port.removeDai(dai);
+        }
+
+        OSCEntityManager.delete(em, port, this.txBroadcastUtil);
+    }
+
+    /**
+     * This method is only called when all the ports are gone.
+     * @param em
+     * @param vm
+     */
+    private void deleteFromDB(EntityManager em, VM vm) {
+        this.log.info("Removing vm " + vm);
+        OSCEntityManager.delete(em, vm, this.txBroadcastUtil);
+    }
+
+    private void deleteFromDB(EntityManager em, Network network) {
```

Comment:
doesnt look like any difference between this and deleteFromDB(EntityManager em, Network network).

Can we just create a method deleteFromDB(EntityManager em, Set<VMPort> ports)?